content/guides/styling-your-components.mdx

```diff
@@ -0,0 +1,522 @@
+      borderRadius: '10px',
+    },
+    variants: {
+      color: {
+        violet: {
+          backgroundColor: 'violet',
+        },
+        gray: {
+          backgroundColor: 'gray',
+        },
+      }
+    },
+    defaultVariants: {
+      color: 'blue'
+    },
+  });
+```
+
+These components can be used like any other Solid component, with type-safe props derived from your variants. 
+For more information on how to use macaron, visit their [documentation](https://macaron.js.org/docs/installation).
+
+## CSS frameworks
+
+CSS frameworks provide pre-styled components and utility classes to speed up development. 
+
+### TailwindCSS
+
+[Tailwind CSS](https://tailwindcss.com/) is an on-demand utility CSS library that integrates seamlessly with Solid as a built-in PostCSS plugin.
+
+#### Installation
+1. Install Tailwind CSS as a development dependency:
+
+```bash
+npm i --save-dev tailwindcss postcss autoprefixer
+
+# or 
+
+pnpm i --save-dev tailwindcss postcss autoprefixer
+
+# or
+
+yarn add --dev tailwindcss postcss autoprefixer
+
+# Initialize
+
+npx tailwindcss init -p
+```
+2. Since TailwindCSS is configuration-driven, after initializing, a `tailwind.config.js` file will be created at the root of your project directory:
+
+```js
+/** @type {import('tailwindcss').Config} */
+module.exports = {
+  content: ["./index.html", "./src/**/*.{js,ts,jsx,tsx}"],
+  theme: {
+    extend: {},
+  },
+  plugins: [],
+};
+```
+
+For a deeper dive into configuration, you can check out the [Tailwind Official Documentation](https://chat.openai.com/#link-to-tailwind-docs).
+
+#### Add Tailwind directives
+
+In your `src/index.css` file, add the following Tailwind directives:
+
+```css
+@tailwind base;
+@tailwind components;
+@tailwind utilities;
+```
+
+These directives inform PostCSS that you're using Tailwind and establish the order of the directives. You can append custom CSS below these directives.
+
+#### Import TailwindCSS
+
+Import your `index.css` file into the root `index.jsx`  or `index.tsx`  file:
+
+```jsx
+import { render } from 'solid-js/web';
+import App from './App';
+import "./index.css";
+
+render(() => <App />, document.getElementById('root') as HTMLElement);
+```
+
+#### Usage
+
+With Tailwind CSS set up, you can now utilize its utility classes. 
+For instance, if you previously had a `Card.css` file, you can replace or remove it:
+
+```
+/* src/components/Card.css */
+/* Remove or replace these styles with Tailwind utility classes */
+```
+Update your components to use Tailwind's utility classes:
+
+```jsx
+/* src/components/Card.jsx */
+function Card() {
+  return (
+    <div class="grid place-items-center min-h-screen">
+      <div class="h-[160px] aspect aspect-[2] rounded-[16px] shadow-[0_0_0_4px_hsl(0_0%_0%_/_15%)]">Hello, world!</div>
+    </div>
+  );
+}
+```
+
+#### Support
+
+For additional assistance, refer to the [Tailwind CSS/Vite integration guide](https://chat.openai.com/#link-to-tailwind-vite-guide).
```

Comment:
Again, this is a ChatGPT link, right? 

You're referring to this, right? https://tailwindcss.com/docs/guides/vite